x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/TriangleTreeReader.java

```diff
@@ -40,7 +45,7 @@ private static boolean visit(
         byte metadata = input.readByte();
         int thisMinX;
         int thisMinY;
-        if ((metadata & 1 << 2) == 1 << 2) { // component in this node is a point
+        if ((metadata & POINT) == POINT) { // component in this node is a point
```

Comment:
Something like `boolean isPoint(metadata)` would be even clearer, but then we're relying on the compiler to inline the code for performance, which it probably does, but you never know.